### PR TITLE
feat: normalize FormValidationOverlay usage in other components

### PR DESCRIPTION
**Pull request by jbadan**
### Description

1. added unit tests for FormValidationOverlay
1. consistent apply new prop `validationOverlayProps` to all components that consume `FormValidationOverlay`: `Checkbox`, `FormInput`, `FormTextArea`, `Select`, `InputGroup`, `SearchInput`, `Select`, `StepInput`
1. unit tests for  `validationOverlayProp` in all consuming components
1. `validationOverlayProp.formMessageProps` also spread to `FormMessage` when used inside component (outside of `ValidationOverlay` - see `Select` or `SearchInput`). This way, if you are expecting `FormMessage` to receive additional classes, it will whether it's inside the dropdown or as it's own popover. This will be useful in nui-widgets-lab for override Gateway theming colors.
1. `_FormValidationOverlay` (not exported to consumers) now spreads additional props to the wrapping `<div>` element via `wrapperProps`. I couldn't use `...rest` because that creates a rabbithole of problems with the way `Popover` is cloning elements. Possibly worth a followup, but is a ton of work and creates a pretty confusing output just for continuity sake.

BREAKING CHANGES:
1. Removed `fd-list--has-message` class from Select's dropdown - it was adding unnecessary padding. It was a relic from before FormMessage rendered inside a Popover
1. `validationClassName` renamed and moved inside of `validationOverlayProps`, use like `validationOverlayProps.className`

There will be at least 2 additional prs for Select - 1 to add the ability to pass custom classes to each component, the second addressing https://github.com/SAP/fundamental-react/issues/1159

### CODE REVIEW

File: src/Forms/Checkbox.js [hunk 1]
```diff
@@ -143,8 +143,15 @@ Please ensure you are either using a visible \`FormLabel\` or an \`aria-label\`
     labelProps: PropTypes.object,
     /** Sets the `name` for the checkbox input */
     name: PropTypes.string,
-    /** Additional classes to apply to validation popover */
-    validationClassName: PropTypes.string,
```
Comment (skvale) [inline src/Forms/Checkbox.js:147]: Should moving this prop to `validationOverlayProps.className` be listed as a breaking change as well?

File: src/Forms/_FormValidationOverlay.test.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+describe('<FormValidationOverlay />', () => {
+    const setup = (props = {}) => mount(
+        <FormValidationOverlay control={(<input onChange={() => {}} value='Test' />)} {...props} />
+    );
+
+    afterEach(() => {
+        document.body.innerHTML = '';
+    });
+
+    const getFormMessage = () => document.body.querySelector('.fd-popover__popper > div > .fd-form-message');
```
Comment (skvale) [inline src/Forms/_FormValidationOverlay.test.js:15]: Just an idea or an alternative, I've seen tests that include
```jsx
// Ensure the popper is accessible with enzyme
jest.mock('react-dom', () => {
    const original = jest.requireActual('react-dom');
    return {
        ...original,
        createPortal: (node) => node,
    };
});
```
so you can still access the elements via `enzyme`

File: src/Forms/Checkbox.js (hunk 1 above)
Comment (jbadan) [reply src/Forms/Checkbox.js:147]: 100%! Good catch

File: src/Forms/_FormValidationOverlay.test.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        test('should show the FormMessage onFocus', async() => {
+            let element = setup({ validationState: { state: 'error', message: 'test message' } });
+
+            await act(async() => {
+                element.find('.fd-popover').simulate('focus');
+            });
+
+            const messageNode = getFormMessage();
+
+            expect(messageNode).toBeDefined();
```
Comment (skvale) [inline src/Forms/_FormValidationOverlay.test.js:103]: I'm pretty sure `expect(null).toBeDefined()` passes.  I f`querySelector` can't find an element it returns `null`, so this expect will always pass.

Comment (jbadan) [reply src/Forms/_FormValidationOverlay.test.js:103]: Well... crap haha bad unit test!

File: src/Forms/_FormValidationOverlay.test.js (hunk 2 above)
Comment (jbadan) [reply src/Forms/_FormValidationOverlay.test.js:15]: I like this, but I'd like to apply it globally in `config/jest/setup.js` so we don't have some unit tests with it/some without. It makes 450 unit tests fail at the moment so I'll probably skip it on this pr 😆

File: src/Forms/_FormValidationOverlay.test.js (hunk 3 above)
Comment (jbadan) [reply src/Forms/_FormValidationOverlay.test.js:103]: The one after this is going to have the same problem - I'll rework both

File: src/Forms/_FormValidationOverlay.test.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                });
+            });
+
+            expect(
+                element.find('FormMessage').getDOMNode().attributes['data-sample'].value
+            ).toBe('Sample');
+        });
+    });
+
+    describe('Interactions', () => {
```
Comment (jbadan) [inline src/Forms/_FormValidationOverlay.test.js:78]: @skvale Looks like you solved this exact problem in tooltip in widgets recently - I copied and applied what you did there to get this working 🎉

**Comment by jbadan** [comment]
I forgot to merge this as a breaking change 😞

**Comment by bcullman** [comment]
@jbadan any chance we would use `git --amend` to fix this?

**Comment by jbadan** [comment]
@bcullman We could try... but that prop was added 1 day before and removed 1 day after. We're most likely the only ones that noticed it came and went 😆

**Pull request merged**